src/test/java/org/fogbowcloud/manager/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -90,577 +84,847 @@
-        when(computeInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
+        Mockito.when(computeInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
        
         this.image = Mockito.mock(Image.class);
-        when(image.getId()).thenReturn(FAKE_IMAGE_ID);
+        Mockito.when(image.getId()).thenReturn(FAKE_IMAGE_ID);
         
-        when(cloudPluginsHolder.getComputePlugin()).thenReturn(computePlugin);
-        when(cloudPluginsHolder.getAttachmentPlugin()).thenReturn(attachmentPlugin);
-        when(cloudPluginsHolder.getNetworkPlugin()).thenReturn(networkPlugin);
-        when(cloudPluginsHolder.getVolumePlugin()).thenReturn(volumePlugin);
-        when(cloudPluginsHolder.getImagePlugin()).thenReturn(imagePlugin);
-        when(cloudPluginsHolder.getComputeQuotaPlugin()).thenReturn(computeQuotaPlugin);
-        when(aaController.getLocalToken(any(FederationUser.class))).thenReturn(new Token());
+        // mocking cloudPluginsHolder to return the correct plugin for each call
+        Mockito.when(cloudPluginsHolder.getComputePlugin()).thenReturn(computePlugin);
+        Mockito.when(cloudPluginsHolder.getAttachmentPlugin()).thenReturn(attachmentPlugin);
+        Mockito.when(cloudPluginsHolder.getNetworkPlugin()).thenReturn(networkPlugin);
+        Mockito.when(cloudPluginsHolder.getVolumePlugin()).thenReturn(volumePlugin);
+        Mockito.when(cloudPluginsHolder.getImagePlugin()).thenReturn(imagePlugin);
+        Mockito.when(cloudPluginsHolder.getComputeQuotaPlugin()).thenReturn(computeQuotaPlugin);
+        Mockito.when(aaController.getLocalToken(Mockito.any(FederationUser.class))).thenReturn(new Token());
         
+        // starting the object we want to test
         this.localCloudConnector = new LocalCloudConnector(aaController, cloudPluginsHolder);
         
     }
 	
+	// test case: Request a compute instance Mockito.when the plugin returns a correct id
 	@Test
 	public void testRequestComputeInstance() throws FogbowManagerException, UnexpectedException {
+		
+		// set up
 		this.order = Mockito.mock(ComputeOrder.class);
-	    when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
-	    when(computePlugin.requestInstance(any(ComputeOrder.class), any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
-		assertEquals(FAKE_INSTANCE_ID, this.localCloudConnector.requestInstance(order));
-		Mockito.verify(computePlugin, times(1)).requestInstance(any(ComputeOrder.class), any(Token.class));
-	    Mockito.verify(volumePlugin, times(0)).requestInstance(any(VolumeOrder.class), any(Token.class));
-	    Mockito.verify(attachmentPlugin, times(0)).requestInstance(any(AttachmentOrder.class), any(Token.class));
-	    Mockito.verify(networkPlugin, times(0)).requestInstance(any(NetworkOrder.class), any(Token.class));
+	    Mockito.when(this.order.getType()).thenReturn(InstanceType.COMPUTE);
+	    Mockito.when(computePlugin.requestInstance(Mockito.any(ComputeOrder.class), Mockito.any(Token.class))).thenReturn(FAKE_INSTANCE_ID);
```

Comment:
Instead of Mockito.any(ComputeOrder.class) in line 147, shouldn't we use the actual order from line 145?

Reply:
If you agree, I think other test cases have the same problem

Reply:
The order has a generic type (Order). We need to cast it to ComputeOrder before. Is it ok?

Reply:
Updating...A think it's not possible. The another argument of method is mocking the token object.

Message:
"when using matchers all arguments have to be provided by matchers"

Reply:
ok